Directory.Build.targets

```diff
@@ -67,14 +67,24 @@
     <PackageReference Include="Microsoft.CodeAnalysis.BannedApiAnalyzers" />
   </ItemGroup>
 
-  <ItemGroup Condition="'$(IsMgmtLibrary)' == 'true'">
+  <ItemGroup Condition="'$(IsMgmtLibrary)' == 'true' and '$(IsTestGenerationSrcProject)' == 'true'">
     <PackageReference Include="Azure.Core" />
     <PackageReference Include="System.Text.Json" />
     <PackageReference Include="System.ClientModel" />
     <PackageReference Include="Azure.ResourceManager" />
 
     <!-- TODO: Review these file references-->
     <Compile Include="$(AzureCoreSharedSources)AzureResourceProviderNamespaceAttribute.cs" LinkBase="Shared" />
+    <!-- Include management shared code-->
+    <Compile Include="$(MSBuildThisFileDirectory)/src/assets/Management.Shared/**/*.cs" Link="Shared/Management/%(RecursiveDir)/%(Filename)%(Extension)" />
+  </ItemGroup>
+
+  <ItemGroup Condition ="'$(IsMgmtLibrary)' == 'true' and '$(IsTestGenerationTestProject)' == 'true'">
+    <PackageReference Include="Azure.Identity" />
```

Comment:
As we can see in sdk-for-net, this is not needed for most test projects, and it is not included in the test needed sharing dependencies in https://github.com/Azure/azure-sdk-for-net/blob/f67a51ae48ae0628733b1cf207d6fec4b0db12ac/eng/Directory.Build.Common.targets#L262-L268.
And when this is needed, we can just add it manually like https://github.com/Azure/azure-sdk-for-net/blob/f67a51ae48ae0628733b1cf207d6fec4b0db12ac/sdk/compute/Azure.ResourceManager.Compute/tests/Azure.ResourceManager.Compute.Tests.csproj#L7
Shall we just remove it for mgmt test projects? it might just work.

Reply:
This is for test SDKs in autorest.csharp. We will regen project files for them every time. So we cannot manually add `Azure.Identity`. It is only autoest.csharp test projects. it is ok to add more project dependency. But for sdks in azure-sdk-for-net, we will not add `Azure.Identity` by default, they need to manually add if needed.

Reply:
OK, sounds reasonable to me